dagmc/dagnav.py

```diff
@@ -129,9 +129,12 @@ def _check_category_and_dimension(self):
         if geom_dimension == -1 and category is None:
             raise ValueError(f"{identifier} has no category or geom_dimension tags assigned.")
 
-    def __eq__(self, other):
+    def __eq__(self, other) -> bool:
         return self.handle == other.handle
 
+    def __hash__(self) -> int:
+        return hash(self.handle)
```

Comment:
I'm wondering if this should instead be the combination of the `handle` and either the `model` object or even the `pymoab.core.Core` instance? It's fairly likely that one could load two models at once and the memory spaces would overlap. (This probably goes for the `__eq__` method as well...).